apps/dashboard/src/components/billing/features.tsx

```diff
@@ -1,272 +1,196 @@
-import { ApiServiceLevelEnum } from '@novu/shared';
+import {
+  ApiServiceLevelEnum,
+  FeatureNameEnum,
+  getFeatureForTierAsBoolean,
+  getFeatureForTierAsText,
+} from '@novu/shared';
 import { Check } from 'lucide-react';
 import { cn } from '../../utils/ui';
 
 enum SupportedPlansEnum {
-  FREE = ApiServiceLevelEnum.FREE,
-  BUSINESS = ApiServiceLevelEnum.BUSINESS,
-  ENTERPRISE = ApiServiceLevelEnum.ENTERPRISE,
+  FREE = 'FREE',
+  PRO = 'PRO',
+  TEAM = 'TEAM',
+  ENTERPRISE = 'ENTERPRISE',
 }
 
+const SupportedPlansEnumToServiceLevelRecord: Record<SupportedPlansEnum, ApiServiceLevelEnum> = {
+  FREE: ApiServiceLevelEnum.FREE_2025,
+  PRO: ApiServiceLevelEnum.PRO,
+  TEAM: ApiServiceLevelEnum.TEAM,
+  ENTERPRISE: ApiServiceLevelEnum.ENTERPRISE,
+};
 type FeatureValue = {
   value: React.ReactNode;
 };
 
 type Feature = {
   label: string;
   isTitle?: boolean;
-  values: {
-    [SupportedPlansEnum.FREE]: FeatureValue;
-    [SupportedPlansEnum.BUSINESS]: FeatureValue;
-    [SupportedPlansEnum.ENTERPRISE]: FeatureValue;
-  };
+  values: Record<SupportedPlansEnum, FeatureValue>;
 };
 
+interface BuildValuesParams {
+  featureName?: FeatureNameEnum;
+  isBoolean?: boolean;
+  prefix?: string | React.ReactNode;
+  suffix?: string | React.ReactNode;
+}
+
+function buildValues(params: BuildValuesParams): Record<SupportedPlansEnum, FeatureValue> {
+  return Object.values(SupportedPlansEnum).reduce(
+    (acc, plan) => {
+      const apiServiceLevel = SupportedPlansEnumToServiceLevelRecord[plan];
+      if (params.isBoolean) {
+        const bool = params.featureName ? getFeatureForTierAsBoolean(params.featureName, apiServiceLevel) : '';
+        acc[plan] = {
+          value: bool ? <Check className="h-4 w-4" /> : '-',
+        };
+        return acc;
+      }
+      const text = params.featureName ? getFeatureForTierAsText(params.featureName, apiServiceLevel) : '';
+      const value = `${params.prefix || ''}${text}${params.suffix || ''}`;
+      acc[plan] = {
+        value,
+      };
+      return acc;
+    },
+    {} as Record<SupportedPlansEnum, FeatureValue>
+  );
+}
+
```

Comment:
This seems like an over-abstraction to me - I would simply leave the frontend plans hardcoded, there don't need to be tied to the shared tier config.